cereus/src/main/java/com/cactus/adapters/AuthLogin.java

```diff
@@ -0,0 +1,186 @@
+package com.cactus.adapters;
+
+import com.cactus.entities.User;
+
+import org.apache.http.client.utils.URIBuilder;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.HashMap;
+
+
+/**
+ * The AuthLogin class implements the AuthAdapter Interface by interacting with a server
+ * that stores our Users and Grocery Lists.
+ */
+public class AuthLogin implements  AuthAdapter{
+
+    /**
+     * Returns a User object that corresponds to the provided username and password.
+     *
+     * The username and password is sent to a server to return the User corresponding to the username and password.
+     *
+     * If the username and password do not correspond to an existing User,
+     * then null is returned, and no login takes place.
+     *
+     * Note: the password is not present in the User, and a null value is in place
+     *
+     * @param username a String containing the username of the user to be logged in
+     * @param password a String containing the password to validate
+     * @return         a User object that corresponds to the given parameters
+     *
+     * @see User
+     */
+    @Override
+    public User login(String username, String password) {
+        HashMap<String, String> login = new HashMap<>();
+        login.put("username", username);
+        login.put("password", password);
+        URI uri;
+        try{
+            uri = new URIBuilder("http://localhost:8080/login").addParameter("username",
+                username).addParameter("password", password).build();
+            }
+        catch(URISyntaxException u) {
+            return null;
+        }
+        User user;
+        try {
+            user = sendRequest(login, uri);
+        }
+        catch(IOException | InterruptedException i){
+            return null;
+        }
+        if (user.getToken() == null) {
```

Comment:
If login fails, a 401 error is returned (I'm not sure if this was mentioned anywhere), or in other words a User object is _not_ returned. However, if login was successful, then a token is _always_ returned. Since HTTP errors are, I'm assuming, caught in the above IOException/InterruptedException block anyways, is this necessary?

Reply:
I check the status code in the helper method sendRequest as the user will have null attributes if the credentials are wrong. This check is now removed.